django/utils/dateformat.py

```diff
@@ -202,6 +205,11 @@ def Z(self):
         # Positive offsets have days=0
         return offset.days * 86400 + offset.seconds
 
+    def r(self):
+        # Implement a stub here so that using the `r` format with a date object
+        # raises the correct error.
```

Comment:
I'm not sure if this comment is necessary.

Reply:
I figured it might be a bit confusing to see a `NotImplemented` there but only for `r()`.

As far as I can tell, `r()` is the only format that strictly accepts datetime objects and not a combination of datetime/date or datetime/time.

Reply:
Agreed, it can be confusing.

Reply:
TBH even now it's confusing :thinking: _"explicit is better than implicit"_ 🧘‍♂️ so I decided to move it to the `DateFormat.r()`.